## Analýza zamestnancov VŠ

In [1]:
import pandas as pd

from pandasql import sqldf
# from sklearn import datasets

In [2]:
df = pd.read_csv('../data/preprocessed/regzam.csv', index_col=0)
display(df)

Employee_ID GivenName  FamilyName DegreePedagogicalAcademic  \
0      18799121690    Štefan       Kočan                      Doc.   
1      14002860472     Dušan       Korgo                       DOC   
2      17922733804       Ján     Buzalka                     prof.   
3      16290288493     Jozef   Stieranka                     prof.   
4      21401303068     Ľuboš       Wäldl                      doc.   
...            ...       ...         ...                       ...   
32725  12110964830      Joža      Spurný                      doc.   
32726  13500390262   Marcela   Harčárová                       NaN   
32727  17108478442      Jana    Juhásová                       NaN   
32728  15917135755   Miriama    Kubušová                       NaN   
32729  51680829014      Jana  Ochodnícka                       NaN   

      DegreeMaster DegreeDoctor  \
0            JUDr.         PhD.   
1             JUDR          PHD   
2            PhDr.         CSc.   
3             Ing.         PhD.   
4            JUDr.         PhD.   
...            ...          ...   
32725   JUDr.PhDr.        Ph.D.   
32726         Mgr.          NaN   
32727         Mgr.          NaN   
32728         Mgr.          NaN   
32729        PhDr.          NaN   

                         Employment_Workplace_University   Employment_Type  \
0      Vysoká škola bezpečnostného manažérstva v Koši...            Docent   
1                                                    NaN               NaN   
2                                                    NaN               NaN   
3                             Akadémia Policajného zboru          Profesor   
4                             Akadémia Policajného zboru          Profesor   
...                                                  ...               ...   
32725  Vysoká škola zdravotníctva a sociálnej práce s...            Docent   
32726  Vysoká škola zdravotníctva a sociálnej práce s...          Asistent   
32727  Vysoká škola zdravotníctva a sociálnej práce s...          Asistent   
32728  Vysoká škola zdravotníctva a sociálnej práce s...          Asistent   
32729  Vysoká škola zdravotníctva a sociálnej práce s...  Odborný asistent   

                                      Employment_Subtype Employment_BeginDate  \
0      s titulom docent a vysokoškolským vzdelaním tr...           2020-10-01   
1                                                    NaN                  NaN   
2                                                    NaN                  NaN   
3      s titulom profesor a vysokoškolským vzdelaním ...           2013-11-01   
4      s titulom docent a vysokoškolským vzdelaním tr...           2008-09-01   
...                                                  ...                  ...   
32725  s titulom docent a vysokoškolským vzdelaním tr...           2022-09-30   
32726          s vysokoškolským vzdelaním druhého stupňa           2022-09-01   
32727          s vysokoškolským vzdelaním druhého stupňa           2022-09-01   
32728          s vysokoškolským vzdelaním druhého stupňa           2022-09-01   
32729         s vysokoškolským vzdelaním tretieho stupňa           2022-09-30   

      Employment_Ratio                             Guarantee_StudyProgram  \
0                  100                                                NaN   
1                  NaN                                                NaN   
2                  NaN                                                NaN   
3                  100  ['bezpečnostnoprávna ochrana osôb a majetku', ...   
4                  100                                                NaN   
...                ...                                                ...   
32725              100                                                NaN   
32726               50                                                NaN   
32727               50                                                NaN   
32728               50                                           

In [3]:
id_set = set(df["Employee_ID"])
print(len(id_set))

32730


-> máme Dataset s 32730 riadkami, pričom žiadne 2 Employee_ID nie sú rovnaké

In [4]:
name_set = set(df["FamilyName"] + " " + df["GivenName"])
len(name_set)

31639

-> 31639 rôznych mien; zostáva zistiť, či pri duplikátnych menách sa jedná o tú istú osobu, alebo nie.
Tiež by nás mohlo zaujímať, ako ukladali osoby, ktorým sa zmenilo meno...

In [5]:
def find_name_duplicates(df):
    name_set = set()
    name_duplicates = set()
    duplicates_ids = set()
    for index, row in df.iterrows():
        name = row["FamilyName"] + " " + row["GivenName"]
        if name in name_set:
            name_duplicates.add(name)
        else:
            name_set.add(name)
    for index, row in df.iterrows():
        name = row["FamilyName"] + " " + row["GivenName"]
        if name in name_duplicates:
            duplicates_ids.add(row["Employee_ID"])
    return tuple(name_duplicates), tuple(duplicates_ids)

In [6]:
name_duplicates, ids = find_name_duplicates(df)

In [7]:
q = f"SELECT * FROM df WHERE Employee_ID IN {ids}"
duplicates_df = sqldf(q, globals()).sort_values(by=['FamilyName', 'GivenName'])
display(duplicates_df)

Employee_ID  GivenName FamilyName DegreePedagogicalAcademic  \
1327  49910822214   Katarína  Adamíková                      None   
1576  18913486096   Katarína  Adamíková                      None   
114   39501978150  František    Adamčík                     prof.   
1325  17240205086  František    Adamčík                      None   
133   51642863772       Jana   Antalová                      None   
...           ...        ...        ...                       ...   
1876  15552653387      Tomáš      Žilka                      None   
1107  10056354153   Viktória    Žilková                      None   
1994  14058627798   Viktória    Žilková                      None   
530   14639882132       Anna    Žuffová                      None   
1296  96699806419       Anna    Žuffová                      None   

     DegreeMaster DegreeDoctor  \
1327         Mgr.         None   
1576         Mgr.         None   
114          Ing.         CSc.   
1325         Ing.         PhD.   
133         MUDr.         PhD.   
...           ...          ...   
1876        MUDr.         None   
1107         Mgr.         None   
1994         None         MSc.   
530          None         None   
1296         Mgr.         None   

                     Employment_Workplace_University  \
1327                                            None   
1576                                            None   
114                 Technická univerzita v Košiciach   
1325                                            None   
133               Univerzita Komenského v Bratislave   
...                                              ...   
1876  Slovenská zdravotnícka univerzita v Bratislave   
1107                                            None   
1994              Univerzita Komenského v Bratislave   
530                                             None   
1296                                            None   

                                        Employment_Type  \
1327                                               None   
1576                                               None   
114                                            Profesor   
1325                                               None   
133                                    Odborný asistent   
...                                                 ...   
1876  Zamestnanec vykonávajúci činnosti VŠ učiteľa m...   
1107                                               None   
1994                     Výskumný pracovník - výskumník   
530                                                None   
1296                                               None   

                                     Employment_Subtype Employment_BeginDate  \
1327                                               None                 None   
1576                                               None                 None   
114   s titulom profesor a vysokoškolským vzdelaním ...           2005-02-01   
1325                                               None                 None   
133          s vysokoškolským vzdelaním tretieho stupňa           2020-08-15   
...                                                 ...                  ...   
1876          s vysokoškolským vzdelaním druhého stupňa           2022-02-01   
1107                                               None                 None   
1994          s vysokoškolským vzdelaním druhého stupňa           2022-06-15   
530                                                None                 None   
1296                                               None                 None   

     Employment_Ratio                             Guarantee_StudyProgram  \
1327             None                                               None   
1576             None                                               None   
114               100  ['prevádzka lietadiel', 'prevádzka lietadiel',...   
1325             None                                               None   
133               100                        

In [8]:
df.query("GivenName == 'Viktória' and FamilyName == 'Žilková'")

Employee_ID GivenName FamilyName DegreePedagogicalAcademic  \
17361  10056354153  Viktória    Žilková                       NaN   
32149  14058627798  Viktória    Žilková                       NaN   

      DegreeMaster DegreeDoctor     Employment_Workplace_University  \
17361         Mgr.          NaN                                 NaN   
32149          NaN         MSc.  Univerzita Komenského v Bratislave   

                      Employment_Type  \
17361                             NaN   
32149  Výskumný pracovník - výskumník   

                              Employment_Subtype Employment_BeginDate  \
17361                                        NaN                  NaN   
32149  s vysokoškolským vzdelaním druhého stupňa           2022-06-15   

      Employment_Ratio Guarantee_StudyProgram Guarantee_University  \
17361              NaN                    NaN                  NaN   
32149               50                    NaN                  NaN   

      Guarantee_Faculty Guarantee_FormOfStudy Guarantee_LevelOfStudy  \
17361               NaN                   NaN                    NaN   
32149               NaN                   NaN                    NaN   

      Employment_Workplace_Faculty Employment_EndDate  
17361                          NaN                NaN  
32149             Lekárska fakulta                NaN

... V tomto bode sme zistili, že naše stiahnuté xml zrejme obsahuje len dáta pre aktuálnych zamestnancov...

Pre minulých (zamestnancov) neobsahuje údaje o zamestnaní, predovšetkým nevieme z dát zistiť, v akom období a kde boli zamestnaní, hoci ich meno je v dátach.

In [9]:
df.query("Employment_EndDate.isna()")

Employee_ID GivenName  FamilyName DegreePedagogicalAcademic  \
0      18799121690    Štefan       Kočan                      Doc.   
1      14002860472     Dušan       Korgo                       DOC   
2      17922733804       Ján     Buzalka                     prof.   
3      16290288493     Jozef   Stieranka                     prof.   
4      21401303068     Ľuboš       Wäldl                      doc.   
...            ...       ...         ...                       ...   
32725  12110964830      Joža      Spurný                      doc.   
32726  13500390262   Marcela   Harčárová                       NaN   
32727  17108478442      Jana    Juhásová                       NaN   
32728  15917135755   Miriama    Kubušová                       NaN   
32729  51680829014      Jana  Ochodnícka                       NaN   

      DegreeMaster DegreeDoctor  \
0            JUDr.         PhD.   
1             JUDR          PHD   
2            PhDr.         CSc.   
3             Ing.         PhD.   
4            JUDr.         PhD.   
...            ...          ...   
32725   JUDr.PhDr.        Ph.D.   
32726         Mgr.          NaN   
32727         Mgr.          NaN   
32728         Mgr.          NaN   
32729        PhDr.          NaN   

                         Employment_Workplace_University   Employment_Type  \
0      Vysoká škola bezpečnostného manažérstva v Koši...            Docent   
1                                                    NaN               NaN   
2                                                    NaN               NaN   
3                             Akadémia Policajného zboru          Profesor   
4                             Akadémia Policajného zboru          Profesor   
...                                                  ...               ...   
32725  Vysoká škola zdravotníctva a sociálnej práce s...            Docent   
32726  Vysoká škola zdravotníctva a sociálnej práce s...          Asistent   
32727  Vysoká škola zdravotníctva a sociálnej práce s...          Asistent   
32728  Vysoká škola zdravotníctva a sociálnej práce s...          Asistent   
32729  Vysoká škola zdravotníctva a sociálnej práce s...  Odborný asistent   

                                      Employment_Subtype Employment_BeginDate  \
0      s titulom docent a vysokoškolským vzdelaním tr...           2020-10-01   
1                                                    NaN                  NaN   
2                                                    NaN                  NaN   
3      s titulom profesor a vysokoškolským vzdelaním ...           2013-11-01   
4      s titulom docent a vysokoškolským vzdelaním tr...           2008-09-01   
...                                                  ...                  ...   
32725  s titulom docent a vysokoškolským vzdelaním tr...           2022-09-30   
32726          s vysokoškolským vzdelaním druhého stupňa           2022-09-01   
32727          s vysokoškolským vzdelaním druhého stupňa           2022-09-01   
32728          s vysokoškolským vzdelaním druhého stupňa           2022-09-01   
32729         s vysokoškolským vzdelaním tretieho stupňa           2022-09-30   

      Employment_Ratio                             Guarantee_StudyProgram  \
0                  100                                                NaN   
1                  NaN                                                NaN   
2                  NaN                                                NaN   
3                  100  ['bezpečnostnoprávna ochrana osôb a majetku', ...   
4                  100                                                NaN   
...                ...                                                ...   
32725              100                                                NaN   
32726               50                                                NaN   
32727               50                                                NaN   
32728               50                                           

## -> Až 32434 riadkov nemá EndDate, čo aj potvrdzuje našu hypotézu, že máme zlé dáta :(  (iba súčasné úväzky)

## **- Pracujú zamestnanci aj na viacerých pracoviskách a vidno v tom nejaké trendy alebo zhluky?**

Vyberieme iba dáta o súčasných zamestnancoch, t.j. odfiltrujeme tie riadky, ktorým chýba BeginDate

In [34]:
current_employees = df[df['Employment_BeginDate'].notna()]
display(current_employees)

Employee_ID  GivenName  FamilyName DegreePedagogicalAcademic  \
0      18799121690     Štefan       Kočan                      Doc.   
3      16290288493      Jozef   Stieranka                     prof.   
4      21401303068      Ľuboš       Wäldl                      doc.   
5      69828748660      Matej     Kostrec                       NaN   
7      91221451321  Magdaléna  Krajníková                       NaN   
...            ...        ...         ...                       ...   
32725  12110964830       Joža      Spurný                      doc.   
32726  13500390262    Marcela   Harčárová                       NaN   
32727  17108478442       Jana    Juhásová                       NaN   
32728  15917135755    Miriama    Kubušová                       NaN   
32729  51680829014       Jana  Ochodnícka                       NaN   

      DegreeMaster DegreeDoctor  \
0            JUDr.         PhD.   
3             Ing.         PhD.   
4            JUDr.         PhD.   
5            JUDr.         PhD.   
7            JUDr.         PhD.   
...            ...          ...   
32725   JUDr.PhDr.        Ph.D.   
32726         Mgr.          NaN   
32727         Mgr.          NaN   
32728         Mgr.          NaN   
32729        PhDr.          NaN   

                         Employment_Workplace_University   Employment_Type  \
0      Vysoká škola bezpečnostného manažérstva v Koši...            Docent   
3                             Akadémia Policajného zboru          Profesor   
4                             Akadémia Policajného zboru          Profesor   
5                             Akadémia Policajného zboru  Odborný asistent   
7                             Akadémia Policajného zboru  Odborný asistent   
...                                                  ...               ...   
32725  Vysoká škola zdravotníctva a sociálnej práce s...            Docent   
32726  Vysoká škola zdravotníctva a sociálnej práce s...          Asistent   
32727  Vysoká škola zdravotníctva a sociálnej práce s...          Asistent   
32728  Vysoká škola zdravotníctva a sociálnej práce s...          Asistent   
32729  Vysoká škola zdravotníctva a sociálnej práce s...  Odborný asistent   

                                      Employment_Subtype Employment_BeginDate  \
0      s titulom docent a vysokoškolským vzdelaním tr...           2020-10-01   
3      s titulom profesor a vysokoškolským vzdelaním ...           2013-11-01   
4      s titulom docent a vysokoškolským vzdelaním tr...           2008-09-01   
5             s vysokoškolským vzdelaním tretieho stupňa           2015-01-01   
7             s vysokoškolským vzdelaním tretieho stupňa           2013-04-01   
...                                                  ...                  ...   
32725  s titulom docent a vysokoškolským vzdelaním tr...           2022-09-30   
32726          s vysokoškolským vzdelaním druhého stupňa           2022-09-01   
32727          s vysokoškolským vzdelaním druhého stupňa           2022-09-01   
32728          s vysokoškolským vzdelaním druhého stupňa           2022-09-01   
32729         s vysokoškolským vzdelaním tretieho stupňa           2022-09-30   

      Employment_Ratio                             Guarantee_StudyProgram  \
0                  100                                                NaN   
3                  100  ['bezpečnostnoprávna ochrana osôb a majetku', ...   
4                  100                                                NaN   
5                  100                                                NaN   
7                  100                                                NaN   
...                ...                                                ...   
32725              100                                                NaN   
32726               50                                                NaN   
32727               50                                                NaN   
32728               50                               

- Kontrola, či majú všetci vybraní zamestnanci EndDate NaN:

In [37]:
current_employees[current_employees['Employment_EndDate'].notna()]

Employee_ID  GivenName FamilyName DegreePedagogicalAcademic  \
260    18814294599     Zuzana   Francová                      doc.   
312    35831820319     Róbert      Rehák                      doc.   
423    10737086394     Renáta   Pakšiová                     prof.   
550    20667634782      Tomáš      Dudáš                      Doc.   
604    86267871612     Danuša    Lišková                       NaN   
...            ...        ...        ...                       ...   
32561  12242815951      Viera   Jakušová                       NaN   
32578  11842718393   Patrícia  Dragunová                       NaN   
32598  56924446213  Vratislav     Zervan                       NaN   
32642  97138493313      Radko   Dvorščák                       NaN   
32673  62384167923      Matúš       Knap                       NaN   

      DegreeMaster DegreeDoctor              Employment_Workplace_University  \
260           Ing.         PhD.           Ekonomická univerzita v Bratislave   
312           Ing.         PhD.           Ekonomická univerzita v Bratislave   
423      Ing. Mgr.         PhD.           Ekonomická univerzita v Bratislave   
550           PhD.         Ing.                     Paneurópska vysoká škola   
604            NaN          NaN      Univerzita sv. Cyrila a Metoda v Trnave   
...            ...          ...                                          ...   
32561         Mgr.         PhD.           Univerzita Komenského v Bratislave   
32578        MUDr.          NaN           Univerzita Komenského v Bratislave   
32598          NaN          NaN      Univerzita sv. Cyrila a Metoda v Trnave   
32642         Ing.          NaN               Prešovská univerzita v Prešove   
32673          Bc.          NaN  Slovenská technická univerzita v Bratislave   

                                         Employment_Type  \
260                                               Docent   
312                                               Docent   
423                                             Profesor   
550                                               Docent   
604    Zamestnanec vykonávajúci činnosti VŠ učiteľa m...   
...                                                  ...   
32561                                   Odborný asistent   
32578                                           Asistent   
32598  Zamestnanec vykonávajúci činnosti VŠ učiteľa m...   
32642  Zamestnanec vykonávajúci činnosti VŠ učiteľa m...   
32673                     Výskumný pracovník - výskumník   

                                      Employment_Subtype Employment_BeginDate  \
260           s vysokoškolským vzdelaním tretieho stupňa           2022-09-01   
312           s vysokoškolským vzdelaním tretieho stupňa           2022-09-01   
423    s titulom docent a vysokoškolským vzdelaním tr...           2021-09-01   
550    s titulom docent a vysokoškolským vzdelaním tr...           2012-05-01   
604                        bez vysokoškolského vzdelania           2022-09-30   
...                                                  ...                  ...   
32561         s vysokoškolským vzdelaním tretieho stupňa           2022-07-07   
32578          s vysokoškolským vzdelaním druhého stupňa           2022-08-01   
32598                      bez vysokoškolského vzdelania           2022-09-30   
32642                      bez vysokoškolského vzdelania           2022-09-22   
32673           s vysokoškolským vzdelaním prvého stupňa           2022-07-01   

      Employment_Ratio                             Guarantee_StudyProgram  \
260                100                                                NaN   
312                100                                                NaN   
423                100                                                NaN   
550                100  ['ekonomika a manažment podnikania', 'ekonomik...   
604                105                                                NaN   
...                ...                           

-> nemajú, sú tam aj zamestnanci, ktorí skončili k 30.9.2022

In [43]:
set(current_employees['Employment_EndDate'])

{'2022-09-30', "['2022-09-30', '2022-09-30']", nan}

-> nechceme ani tých s EndDate 30.9.2022

In [49]:
current_employees = current_employees[current_employees['Employment_EndDate'].isna()].reset_index()

- Majú všetci University vyplnenú?

In [53]:
current_employees[current_employees['Employment_Workplace_University'].isna()]

Empty DataFrame
Columns: [index, Employee_ID, GivenName, FamilyName, DegreePedagogicalAcademic, DegreeMaster, DegreeDoctor, Employment_Workplace_University, Employment_Type, Employment_Subtype, Employment_BeginDate, Employment_Ratio, Guarantee_StudyProgram, Guarantee_University, Guarantee_Faculty, Guarantee_FormOfStudy, Guarantee_LevelOfStudy, Employment_Workplace_Faculty, Employment_EndDate]
Index: []

-> áno.

Existujú ľudia s viacerými záznamami? Napr., existuje viacero riadkov s rovnakým indexom?

In [56]:
len(set(current_employees['Employee_ID']))

13735

=> Nie, každý riadok má svoj vlastný Employee_ID

In [84]:
counts = current_employees[['Employee_ID', 'GivenName', 'FamilyName']].groupby(['GivenName', 'FamilyName']).count().rename(columns={'Employee_ID': 'count'}).reset_index()
display(counts)

GivenName  FamilyName  count
0          Vasil    Gluchman      1
1          Aaron    McCauley      1
2      Abdolreza      Majidi      1
3           Adam        Antl      1
4           Adam       Bžoch      1
...          ...         ...    ...
13500      Žofia   Bárcziová      1
13501      Žofia      Dubová      1
13502      Žofia  Kuzevičová      1
13503      Žofia   Nižnanská      1
13504      Žofia    RÁDIKOVÁ      1

[13505 rows x 3 columns]

-> títo ľudia majú viacero záznamov alebo rovnaké mená:

In [87]:
counts[counts['count'] != 1]

GivenName   FamilyName  count
187        Alena      Nagyová      2
314    Alexandra  Valenčáková      2
448       Andrea       Gažová      2
497       Andrea    Miškufová      2
546       Andrea      Vargová      2
...          ...          ...    ...
13014     Zuzana   Štefunková      2
13165    Ľubomír       Straka      2
13238      Ľuboš       Gajdoš      2
13331    Ľudovít      Danihel      2
13456     Štefan         Tóth      2

[209 rows x 3 columns]

In [90]:
multi_employees = counts[counts['count'] != 1]
multi_employees[['GivenName', 'FamilyName']]

GivenName   FamilyName
187        Alena      Nagyová
314    Alexandra  Valenčáková
448       Andrea       Gažová
497       Andrea    Miškufová
546       Andrea      Vargová
...          ...          ...
13014     Zuzana   Štefunková
13165    Ľubomír       Straka
13238      Ľuboš       Gajdoš
13331    Ľudovít      Danihel
13456     Štefan         Tóth

[209 rows x 2 columns]

In [86]:
current_employees[current_employees['FamilyName'] == 'Výrost']

index  Employee_ID GivenName FamilyName DegreePedagogicalAcademic  \
324    704  53060727347     Tomáš     Výrost                      doc.   

    DegreeMaster DegreeDoctor  \
324         Ing.         PhD.   

                       Employment_Workplace_University       Employment_Type  \
324  ['Ekonomická univerzita v Bratislave', 'Techni...  ['Docent', 'Docent']   

                                    Employment_Subtype  \
324  ['s titulom docent a vysokoškolským vzdelaním ...   

             Employment_BeginDate Employment_Ratio Guarantee_StudyProgram  \
324  ['2021-03-01', '2021-01-04']    ['100', '50']                    NaN   

    Guarantee_University Guarantee_Faculty Guarantee_FormOfStudy  \
324                  NaN               NaN                   NaN   

    Guarantee_LevelOfStudy                Employment_Workplace_Faculty  \
324                    NaN  ['Obchodná fakulta', 'Ekonomická fakulta']   

    Employment_EndDate  
324                NaN

## Ešte raz z upravených dát:
    

In [110]:
df_workplaces = pd.read_csv('../data/preprocessed/regzam_workplaces.csv', index_col=0)
display(df_workplaces)

Employee_ID FamilyName     GivenName DegreePedagogicalAcademic  \
0      18799121690      Kočan        Štefan                      Doc.   
1      14002860472      Korgo         Dušan                       DOC   
2      17922733804    Buzalka           Ján                     prof.   
3      16290288493  Stieranka         Jozef                     prof.   
4      21401303068      Wäldl         Ľuboš                      doc.   
...            ...        ...           ...                       ...   
33358  17628642100      Peter          Fabo                       NaN   
33359  17628642100      Peter          Fabo                       NaN   
33360  28653015119   Marianna  Berinšterová                       NaN   
33361  28653015119   Marianna  Berinšterová                       NaN   
33362  28653015119   Marianna  Berinšterová                       NaN   

      DegreeMaster DegreeDoctor  \
0            JUDr.         PhD.   
1             JUDR          PHD   
2            PhDr.         CSc.   
3             Ing.         PhD.   
4            JUDr.         PhD.   
...            ...          ...   
33358        RNDr.         PhD.   
33359        RNDr.         PhD.   
33360         Mgr.         PhD.   
33361         Mgr.         PhD.   
33362         Mgr.         PhD.   

                         Employment_Workplace_University  \
0      Vysoká škola bezpečnostného manažérstva v Koši...   
1                                                    NaN   
2                                                    NaN   
3                             Akadémia Policajného zboru   
4                             Akadémia Policajného zboru   
...                                                  ...   
33358                       Žilinská univerzita v Žiline   
33359  Trenčianska univerzita Alexandra Dubčeka v Tre...   
33360                     Prešovská univerzita v Prešove   
33361  Vysoká škola medzinárodného podnikania ISM Slo...   
33362                     Prešovská univerzita v Prešove   

                Employment_Workplace_Faculty  \
0                                        NaN   
1                                        NaN   
2                                        NaN   
3                                        NaN   
4                                        NaN   
...                                      ...   
33358                                    NaN   
33359            Fakulta špeciálnej techniky   
33360  Fakulta humanitných a prírodných vied   
33361                                    NaN   
33362                         Fakulta športu   

                                         Employment_Type  Employment_Ratio  
0                                                 Docent            100.00  
1                                                    NaN               NaN  
2                                                    NaN               NaN  
3                                               Profesor            100.00  
4                                               Profesor            100.00  
...                                                  ...               ...  
33358                     Výskumný pracovník - výskumník             20.00  
33359                     Výskumný pracovník - výskumník            100.00  
33360                                   Odborný asistent            100.00  
33361  Zamestnanec vykonávajúci činnosti VŠ učiteľa m...              5.00  
33362  Zamestnanec vykonávajúci činnosti VŠ učiteľa m...             15.55  

[33363 rows x 10 columns]